### PR TITLE
Support  changing language on nav bar button

### PR DESCRIPTION
**Pull request by miurahr**
Add logic to change language.
Call for designer to change placement.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

**Comment by SethMichaelLarson** [comment]
Wouldn't it be best to have the language selector always be in it's native language?

**Comment by miurahr** [comment]
> Wouldn't it be best to have the language selector always be in it's native language? (ie 'Japanese' shows as '日' even on the English version of the site.)

It would be better but it is out-of-scope of localization. So we can not use i18n framework for this.
We need to have a scheme to convert two-word language code (or 4 word code) to  local language name in local languages.

for example,

en - English
ja -  日本語
zh_CN -  汉语
zh_HK - 中文繁體
fr - français

Where is a complete language code to local name map?

**Comment by miurahr** [comment]
Here may be what we look for;
http://www.geonames.de/languages.html

Here is a ruby module but only provide English names.
https://github.com/scsmith/language_list

**Comment by miurahr** [comment]
Implemented.
Because I'm not good at web design,  I'd like to call for help.

**Comment by miurahr** [comment]
You may like config/languages.yml which  have major language names in every local languages.
When new translation added, and update `available_locales`, automatically show it  as an item on the language selector in local language.

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -116,3 +116,22 @@ img.pquote-avatar {
...
+  background-color: #f9f9f9;
+  min-width: 30px;
+  box-shadow: 0px 2px 2px 0px rgba(0,0,0,0.2);
+  padding: 2px 2px;
+  z-index: 1;
+}
+
+.dropdown:hover .dropdown-content {
+  display: block;
+}
```
Comment (MikeMcQuaid) [inline app/assets/stylesheets/application.scss]: I'd suggest using `http://primercss.io/archive/forms/#selects` for this

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -2,4 +2,8 @@ module ApplicationHelper
   def nbsp_last_word(string)
     string.sub(/ ([a-z]+\.?)$/, "&nbsp;\\1").html_safe
   end
+
+  def language_name(l)
```
Comment (MikeMcQuaid) [inline app/helpers/application_helper.rb]: `def language_name(language)`

File: app/views/layouts/_nav.html.erb [hunk 3]
```diff
@@ -29,4 +29,11 @@
       <%=t "Sign up" %>
     <% end %>
   <% end %>
+    <span class="d-sm-inline-block ml-4 border-right dropdown"><%=t "nav.language.title" %>
```
Comment (MikeMcQuaid) [inline app/views/layouts/_nav.html.erb]: I'd suggest using `http://primercss.io/archive/forms/#selects` for this

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -4,3 +4,8 @@ test:
 
 MikeMcQuaid:
   github_username: "MikeMcQuaid"
+  language: "en"
+
+Miurahr:
+  github_username: "miurahr"
+  language: "ja"
```
Comment (MikeMcQuaid) [inline test/fixtures/users.yml]: Needs a newline at end of file

File: config/locales/en.yml [hunk 5]
```diff
@@ -35,6 +35,10 @@ en:
   # app/views/layouts/_nav.html.erb
   Sign up: "Sign up"
   Sign out: "Sign out"
+  nav:
+    # language selecter
+    language:
+      title: "LANG"
```
Comment (MikeMcQuaid) [inline config/locales/en.yml]: I'd suggest just displaying `English` here.

File: config/locales/ja.yml [hunk 6]
```diff
@@ -34,7 +34,10 @@ ja:
   # app/views/layouts/_nav.html.erb
   Sign up: "ユーザ登録"
   Sign out: "サインアウト"
-  
+  nav:
+    language:
+      title: "言語"
```
Comment (MikeMcQuaid) [inline config/locales/ja.yml]: I'd suggest this be the Japanese word for "Japanese" here.

**Comment by miurahr** [comment]
Because I'm not front end engineer, and here is my best. I don't want to do further for front end.
If someone want to improve it, please pick here and do it.

File: app/views/layouts/_nav.html.erb (hunk 3 above)
Comment (miurahr) [reply app/views/layouts/_nav.html.erb]: Pls you do.

File: test/fixtures/users.yml (hunk 4 above)
Comment (miurahr) [reply test/fixtures/users.yml]: Done.

**Comment by MikeMcQuaid** [comment]
> Because I'm not front end engineer, and here is my best. I don't want to do further for front end.
> If someone want to improve it, please pick here and do it.

Sure thing @miurahr, I'll push to this PR when I get the chance.

**Comment by MikeMcQuaid** [comment]
@miurahr And: thanks again for all your help ❤️

**Comment by MikeMcQuaid** [comment]
Thanks again @miurahr, great work!

**Pull request merged**